### PR TITLE
Create middleware.de.md | German Translation for middleware.md (#751)

### PR DESCRIPTION
**Pull request by nptr23**
translation of the whole document

### CODE REVIEW

**Comment by nptr23** [comment]
Hey, I wanted to contribute to an open source project for the first time. I chose document out of your list, that needed german translation. I hope i did everything correctly.

**Comment by 0xTim** [comment]
@nptr23 thanks! I'll let one of the German speakers review it

File: docs/advanced/middleware.de.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Middleware
+
+Middleware ist eine Logikkette zwischen dem Client und einem Vapor-Route-Handler. Es ermöglicht Ihnen, Operationen an eingehenden Requests durchzuführen, bevor sie den Route-Handler erreichen, und an ausgehenden Responses, bevor sie zum Client gesendet werden.
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:3]:
```suggestion
Middleware ist eine Logikkette zwischen dem Client und einem Vapor-Route-Handler. Es ermöglicht dir, Operationen an eingehenden Requests durchzuführen, bevor sie den Route-Handler erreichen, und an ausgehenden Responses, bevor sie zum Client gesendet werden.
```

File: docs/advanced/middleware.de.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+## Konfiguration
+
+Middleware kann global (für jede Route) in `configure(_:)` mit `app.middleware` registriert werden.
+
+```swift
+app.middleware.use(MyMiddleware())
+```
+
+Middleware kann auch zu idividuellen Routen hinzugefügt werden, in dem Route-Gruppen verwedet werden.
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:13]:
```suggestion
Middleware kann auch zu individuellen Routen hinzugefügt werden, in dem Route-Gruppen verwendet werden.
```

File: docs/advanced/middleware.de.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+```swift
+let group = app.grouped(MyMiddleware())
+group.get("foo") { req in
+	// This request has passed through MyMiddleware.
+}
+```
+
+### Reihenfolge
+
+Die Reihenfolge, in der Middleware hinzugefügt wird, ist wichtig. Requests, die in Ihre Anwendung kommen, durchlaufen die Middleware in der Reihenfolge, in der sie hinzugefügt werden. Responses, die Ihre Anwendung verlassen, gehen in umgekehrter Reihenfolge durch die Middleware zurück. Route-spezifische Middleware wird immer nach der Application-Middleware ausgeführt. Folgendes Beispiel erklärt den Sachverhalt:
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:24]:
```suggestion
Die Reihenfolge, in der Middleware hinzugefügt wird, ist wichtig. Requests, die in die Application eingehen, durchlaufen die Middleware in der Reihenfolge in der sie hinzugefügt werden. Responses, die die Application verlassen, gehen in umgekehrter Reihenfolge durch die Middleware zurück. Route-spezifische Middleware wird immer nach der Application-Middleware ausgeführt. Folgendes Beispiel erklärt den Sachverhalt:
```

File: docs/advanced/middleware.de.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+app.middleware.use(MiddlewareB())
+
+app.group(MiddlewareC()) {
+	$0.get("hello") { req in
+		"Hello, middleware."
+	}
+}
+```
+
+Ein Request an GET /hello wird die Middleware in der folgenden Reihenfolge durchlaufen:
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:37]:
```suggestion
Ein Request an `GET /hello` wird die Middleware in der folgenden Reihenfolge durchlaufen:
```

File: docs/advanced/middleware.de.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+```
+Request → A → B → C → Handler → C → B → A → Response
+```
+
+Middleware kann auch _vorangestellt_ werden, was nützlich ist, wenn Sie eine Middleware _vor_ der Standard-Middleware hinzufügen möchten, die Vapor automatisch hinzufügt:
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:43]:
```suggestion
Middleware kann auch _vorangestellt_ werden, was nützlich ist, wenn du eine Middleware _vor_ der Standard-Middleware hinzufügen möchtest, die Vapor automatisch hinzufügt:
```

File: docs/advanced/middleware.de.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    func respond(to request: Request, chainingTo next: Responder) -> EventLoopFuture<Response> {
+        guard let user = request.auth.get(User.self), user.role == .admin else {
+            return request.eventLoop.future(error: Abort(.unauthorized))
+        }
+        return next.respond(to: request)
+    }
+}
+```
+
+Wenn du async/await verwendest, verwende folgendes Beispiel:
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:72]:
```suggestion
Wenn du `async`/`await` verwendest, verwende folgendes Beispiel:
```

File: docs/advanced/middleware.de.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    func respond(to request: Request, chainingTo next: Responder) -> EventLoopFuture<Response> {
+        next.respond(to: request).map { response in
+            response.headers.add(name: "My-App-Version", value: "v2.5.9")
+            return response
+        }
+    }
+}
+```
+
+Wenn du async/await verwendest, verwende folgendes Beispiel:
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:102]:
```suggestion
Wenn du `async`/`await` verwendest, verwende folgendes Beispiel:
```

File: docs/advanced/middleware.de.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+## File Middleware
+
+FileMiddleware ermöglicht das Ausliefern von Assets aus dem Public-Ordner deines Projekts an den Client. Hier könnten statische Dateien wie Stylesheets oder Bitmap-Bilder enthalten sein.
+
+```swift
+let file = FileMiddleware(publicDirectory: app.directory.publicDirectory)
+app.middleware.use(file)
+```
+
+Sobald FileMiddleware registriert ist, kann eine Datei wie Public/images/logo.png von einer Leaf-Vorlage aus verlinkt werden als <img src="/images/logo.png"/>.
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:125]:
```suggestion
Sobald FileMiddleware registriert ist, kann eine Datei wie `Public/images/logo.png` von einer Leaf-Vorlage aus verlinkt werden als `<img src="/images/logo.png"/>`.
```

File: docs/advanced/middleware.de.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+Eine Beispielkonfiguration könnte folgendermaßen aussehen:
+
+```swift
+let corsConfiguration = CORSMiddleware.Configuration(
+    allowedOrigin: .all,
+    allowedMethods: [.GET, .POST, .PUT, .OPTIONS, .DELETE, .PATCH],
+    allowedHeaders: [.accept, .authorization, .contentType, .origin, .xRequestedWith, .userAgent, .accessControlAllowOrigin]
+)
+let cors = CORSMiddleware(configuration: corsConfiguration)
+// cors middleware should come before default error middleware using `at: .beginning`
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:148]:
```suggestion
// cors-Middleware sollte vor der Standard-Fehler-Middleware mit `at: .beginning` stehen
```

File: docs/advanced/middleware.de.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+app.middleware.use(cors, at: .beginning)
+```
+
+Da geworfene Fehler sofort an den Client zurückgegeben werden, muss die CORSMiddleware vor der ErrorMiddleware aufgeführt werden. Andernfalls wird die HTTP-Fehlerantwort ohne CORS-Header zurückgegeben und kann nicht vom Browser gelesen werden.
```
Comment (ptoffy) [inline docs/advanced/middleware.de.md:152]:
```suggestion
Da geworfene Fehler sofort an den Client zurückgegeben werden, muss die `CORSMiddleware` vor der `ErrorMiddleware` aufgeführt werden. Andernfalls wird die HTTP-Fehlerantwort ohne CORS-Header zurückgegeben und kann nicht vom Browser gelesen werden.
```

**Review by ptoffy — CHANGES_REQUESTED** [review]

**Review by ptoffy — APPROVED** [review]

**Pull request merged**